corgidrp/l2b_to_l3.py

```diff
@@ -19,13 +20,32 @@ def divide_by_exptime(input_dataset):
     
     Divide the data by the exposure time to get the units in electrons/s
 
-    TODO: Make sure to update the headers to reflect the change in units
-
     Args:
         input_dataset (corgidrp.data.Dataset): a dataset of Images (L2b-level)
 
     Returns:
         corgidrp.data.Dataset: a version of the input dataset with the data in units of electrons/s
     """
 
-    return None
+    data = input_dataset.copy()
+
+    all_data_new = np.zeros(data.all_data.shape)
+    all_err_new = np.zeros(data.all_err.shape)
+
+    for i in range(len(data.frames)):
+        exposure_time = float(data.frames[i].ext_hdr['EXPTIME']) #ask if it is the same 
+
+        data.frames[i].data = data.frames[i].data / exposure_time
+        data.frames[i].err = data.frames[i].err / exposure_time
+
+        all_data_new[i] = data.frames[i].data
+        all_err_new[i] = data.frames[i].err
```

Comment:
If I understood correctly, your suggestion is to update the data.frames.data and extract the data.all_data array. Is this what you mean? The data.frames.data (and .err) are independent from data.all_data (and .all_err), so I think I have to update them separately

Reply:
I think the way the code is currently written is to enable the use of the `update_after_processing_step` function, which is our preferred way to update the object. Is that the case Lisa?

Reply:
Yes, exactly

Reply:
What I meant is, would it be possible to update the data set this way: data.update_after_processing_step(history_msg, new_all_data=data.all_data, new_all_err = data.all_err), 
without the need for creating the "all_data_new" and "all_err_new" arrays and making the code slightly more efficient?

Reply:
I think either solution is adequate here, given that this won't be a resource or time liming step. We're probably talking about ms-level time savings. I actually like Lisa's implementation since it explicitly sets a variable (`all_data_new`) and then uses it to set the new data in `update_after_processing_step`. There's a slight danger/opaqueness of setting `data.frames[i].data` and then using `data.all_data` to pass into `update_after_processing_step`. There's an implicit assumption those two variables are views of the same data, which can be open to bugs when they aren't actually pointing to the same piece of memory as well as making the code slightly less opaque.

Either way, I think the implementation is probably good enough for us!

Reply:
@JuergenSchreiber, you are referring to Marie's suggestion? I think you might be right, although there should be some quick fixes that would make it work again. Either way, I think we shouldn't spend too much time on analyzing this, unless folks are really interested in coming up with the most succinct solution.